src/NzbDrone.Core/Download/Clients/rTorrent/RTorrentProxy.cs

```diff
@@ -253,45 +188,86 @@ private string[] GetCommands(string label, RTorrentPriority priority, string dir
             return result.ToArray();
         }
 
-        private IRTorrent BuildClient(RTorrentSettings settings)
+        private HttpRequestBuilder BuildRequest(RTorrentSettings settings)
         {
-            var client = XmlRpcProxyGen.Create<IRTorrent>();
-
-            client.Url = string.Format(@"{0}://{1}:{2}/{3}",
-                                    settings.UseSsl ? "https" : "http",
-                                    settings.Host,
-                                    settings.Port,
-                                    settings.UrlBase);
-
-            client.EnableCompression = true;
+            var requestBuilder = new HttpRequestBuilder(settings.UseSsl, settings.Host, settings.Port, settings.UrlBase)
+            {
+                LogResponseContent = true,
+            };
 
             if (!settings.Username.IsNullOrWhiteSpace())
             {
-                client.Credentials = new NetworkCredential(settings.Username, settings.Password);
+                requestBuilder.NetworkCredential = new NetworkCredential(settings.Username, settings.Password);
             }
 
-            return client;
+            return requestBuilder.Post();
         }
 
-        private T ExecuteRequest<T>(Func<T> task)
+        private XElement GetValueElement(object arg)
         {
-            try
+            XElement value;
+
+            if (arg is string s)
             {
-                return task();
+                value = new XElement("string", s);
             }
-            catch (XmlRpcServerException ex)
+            else if (arg is byte[] bytes)
             {
-                throw new DownloadClientException("Unable to connect to rTorrent, please check your settings", ex);
+                value = new XElement("base64", Convert.ToBase64String(bytes));
             }
-            catch (WebException ex)
+            else
             {
-                if (ex.Status == WebExceptionStatus.TrustFailure)
-                {
-                    throw new DownloadClientUnavailableException("Unable to connect to rTorrent, certificate validation failed.", ex);
-                }
+                throw new InvalidOperationException($"Unhandled argument type {arg.GetType().Name}");
+            }
+
+            return new XElement("value", value);
+        }
+
+        private string GenerateRequest(string methodName, params object[] args)
```

Comment:
Does it help us any at all to dump some of this common request generation xmlrpc code in a request builder similar to our `JsonRpcRequestBuilder` in `NzbDrone.Common.Http`?